csharp/src/Ice/Communicator.cs

```diff
@@ -983,9 +983,9 @@ public void Destroy()
             }
 
             {
-                if (Logger != null && Logger is FileLogger)
+                if (Logger != null && Logger is FileLogger fileLogger)
```

Comment:
The first check for `Logger != null` is redundant here. `is` will evaluate to false for null expressions anyways,

Reply:
fixed